RELEASING.md

```diff
@@ -1,18 +1,78 @@
-# Release Process
-
-This doc covers the release process for the functions in the
-kpt-functions-catalog repo.
-
-1. Checking the CI badge status on top of the repo's README page. If the CI is
-   failing on the master, we need to fix it before doing a release.
-1. Go to the [releases pages] in your browser.
-1. Click `Draft a new release` to create a new release for a function. The tag
-   version format should be `{language}/{function-name}/{semver}`. e.g.
-   `go/set-namespace/v1.2.3` and `ts/kubeval/v2.3.4`. The release name should be
-   `{funtion-name} {semver}`. The release notes for this function should be in
-   the body. 
-1. Click `Publish release` button.
+# Release
+
+## Branching Strategy
+
+We maintain a release branch for every minor version of every function.
+
+For a specific major version, only the release branch of the latest minor
+version is in active mode (bugs will be fixed and documentations will be
+updated), while the release branches for the other minor versions are in
+maintenance mode (only critical bugs will be fixed).
+
+## Release Process
+
+### Release Major or Minor Versions
+
+1. Check the CI badge status on top of the repo's README page. If the CI is
+   failing on the master, we must fix it before doing a release.
+1. Create a new release branch from the `master` branch. The new release branch
+   name should be `<function-name>/v<major>.<minor>`. For example, if you want
+   to release version `v1.2.0` for the `gatekeeper` function, you need to create
+   branch `gatekeeper/v1.2`.
+1. Create a PR to update the usage doc and examples
```

Comment:
What if there is a release branch already and I want to update the patch version ? The release branch is not in sync with the master and creates huge diff including all commits from master branch.